File: solutions/csharp/raindrops/1/Raindrops.cs
using System.Collections.Generic;
using System.Linq;
using System.Text;

public static class Raindrops
{
	private static readonly Dictionary<int, string> _textByValue = new()
	{
		{ 3, "Pling" },
		{ 5, "Plang" },
		{ 7, "Plong" },
	};

	public static string Convert(int number)
	{
		string sound = _textByValue
			.Aggregate(
				  new StringBuilder(),
				  (stringBuilder, keyValuePair) =>
					{
						(int value, string text) = keyValuePair;
						return (number % value == 0)
							? stringBuilder.Append(text)
							: stringBuilder;
					})
			.ToString();

		return string.IsNullOrEmpty(sound)
			? number.ToString()
			: sound;
	}
}

File: solutions/python/raindrops/1/raindrops.py
DIVISOR_FOR_PLING = 3
DIVISOR_FOR_PLANG = 5
DIVISOR_FOR_PLONG = 7

SOUND_FOR_PLING = "Pling"
SOUND_FOR_PLANG = "Plang"
SOUND_FOR_PLONG = "Plong"

def convert(number):
    result = ""
    if number % DIVISOR_FOR_PLING == 0:
        result += SOUND_FOR_PLING

    if number % DIVISOR_FOR_PLANG == 0:
        result += SOUND_FOR_PLANG

    if number % DIVISOR_FOR_PLONG == 0:
        result += SOUND_FOR_PLONG

    if not result:
        return str(number)

    return result